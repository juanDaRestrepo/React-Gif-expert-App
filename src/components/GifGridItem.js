import React from 'react'

export const GifGridItem = ({id,url,title}) => {
    
    return (
        <div className='card animate__animated animate__backInLeft '>
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}
