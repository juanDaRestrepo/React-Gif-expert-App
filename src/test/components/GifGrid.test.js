import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");
describe('Pruebas en el componente GifGrid', () => {
    const category = 'One Punch';

    test('debe mostrarse el componente correctamente', () => { 
        useFetchGifs.mockReturnValue({
            data: [],
            loading:true
        })
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostar items cuando se cargan imagenes useFetchGifs', () => {
          const gifs = [{
            id: 'abc',
            url: 'https//localhost/cualquier/cosa.jpg',
            title:'cualquier cosa'
          }];

          useFetchGifs.mockReturnValue({
              data:gifs,
              loading:false
          });

          const wrapper = shallow(<GifGrid category={category}/>)
          expect(wrapper).toMatchSnapshot();
          expect(wrapper.find('p').exists()).toBe(false);
          expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    


    
})