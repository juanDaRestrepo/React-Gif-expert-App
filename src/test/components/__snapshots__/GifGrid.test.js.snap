// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid debe de mostar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="cualquier cosa"
      url="https//localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componente GifGrid debe mostrarse el componente correctamente 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading..
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
